### DISCUSSION TITLE
LM Studio to Support Local Development

### DISCUSSION THREAD

**Discussion by soetedja** [tryAGI/LangChain#417] [Ideas]
Can we currently use LM Studio with LangChain.NET for local embedding, or do we need to develop a new provider to support it?

**Comment by HavenDV** [comment]
Not sure, haven't used it. If there is compatibility with OpenAI SDK, you can just use OpenAiProvider with custom endpoint and modelId and it will work

**Reply by soetedja** [reply]
Will try, thanks!

**Reply by soetedja** [reply]
@HavenDV when we init the OpenAiProvider like this
`
        var provider = new OpenAiProvider("lm-studio", "http://localhost:1234");` 

Do you know why the embedding host is empty? 

I got an error below. 

> No such host is known. (http:443)

I appreciate any help you can provide.

**Reply by HavenDV** [reply]
Try to use latest preview version. There was a transition to the new OpenAI SDK, and now the format with the full address is used, but previously there was a format without the url scheme of the port.
